test/unit/range/iterator_test_template.hpp

```diff
@@ -685,35 +685,43 @@ inline void address_difference_test(it_begin_t it_begin, it_sentinel_t it_end)
 
 TYPED_TEST_P(iterator_fixture, address_of_end)
 {
-    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag>)
+    // sentinel must not be the same as the iterator
+    if constexpr (std::ranges::common_range<decltype(this->test_range)>)
     {
-        using difference_t = std::iter_difference_t<decltype(std::ranges::begin(this->test_range))>;
-        auto it_begin = std::ranges::begin(this->test_range);
-        auto it_end = std::ranges::end(this->test_range);
-        EXPECT_EQ(std::to_address(it_end), std::to_address(it_begin) + static_cast<difference_t>(it_end - it_begin));
-    }
+        if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::contiguous_iterator_tag>)
```

Comment:
The next `if constexpr` uses the same expression and additionally requires `const_iterable`. Can we simplify this such that we do not check for the `std::contiguous_iterator_tag` twice?

Reply:
Same for `address_difference`.

I think refactoring would be more clear, otherwise you start to wonder why it's the same expression (i.e., "Can't I read?", "Is this an error?")